go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -169,25 +177,32 @@ func (bh *S3BackupHandle) AddFile(ctx context.Context, filename string, filesize
 
 	go func() {
 		defer bh.waitGroup.Done()
-		uploader := s3manager.NewUploaderWithClient(bh.client, func(u *s3manager.Uploader) {
+		uploader := manager.NewUploader(bh.client, func(u *manager.Uploader) {
 			u.PartSize = partSizeBytes
 		})
 		object := objName(bh.dir, bh.name, filename)
 		sendStats := bh.bs.params.Stats.Scope(stats.Operation("AWS:Request:Send"))
 		// Using UploadWithContext breaks uploading to Minio and Ceph https://github.com/vitessio/vitess/issues/14188
-		_, err := uploader.Upload(&s3manager.UploadInput{
+		_, err := uploader.Upload(context.Background(), &s3.PutObjectInput{
```

Comment:
Using `context.Background()` is a side-effect of the issue described in https://github.com/vitessio/vitess/issues/14188. Upgrading to the SDK V2 does not fix it, the issue is rooted in how we handle the context and the goroutines. I started investigating the issue a bit deeper and will continue in a subsequent PR.